app/src/views/private/components/drawer-item.vue

```diff
@@ -411,7 +411,11 @@ function useActions() {
 			internalEdits.value[primaryKeyField.value.field] = props.primaryKey;
 		}
 
-		emit('input', internalEdits.value);
+		const noNullDefaultValues = Object.fromEntries(
+			Object.entries(defaultValues.value).filter(([_, value]) => value !== null)
+		);
+
+		emit('input', assign({}, noNullDefaultValues, internalEdits.value));
```

Comment:
This fix perfectly resolves default value of `false` and make sure they are being emitted! 👍 however I believe we need to additionally check the Create permission and filter out `defaultValues.value` from having any fields that a user does not have permission to create.

## Example setup

1. Create collection `a` and collection `b`
2. In collection `b`, create input/string field `name` with default value `Default Name B`
3. In collection `b`, create toggle/boolean field `enabled` with default value `false`
4. In collection `a`, create O2M field to collection `b`
5. Create a new role, grant the role the following permissions:
    - Collection `a`: full Create + Read + Update permissions
    - Collection `b`
        - Create permissions on all fields _except_ `name`
        - Read permissions on all fields
        - Update permissions on all fields _except_ `name`
6. Use the new role to create a new item in collection `a` and use the O2M field
        
    Example screenshot:
    ![](https://user-images.githubusercontent.com/42867097/224365473-b92cf70f-d76f-4a0a-8102-fb70b17ba1f7.png)

## Screen Capture

The B item is created, but it is sending `{ name: "Default Name B", enabled: false }`, whereas the current user should only send `{ enabled: false }` because they do not have create permission on `name`.

https://user-images.githubusercontent.com/42867097/224366154-8317819a-9d4e-4ed6-8a4a-ecd19c1c4500.mp4

Reply:
Good catch, we should probably handle it the same way as the v-form does by only sending over values that are not the default, meaning, we shouldn't inject default vaules into the relational edits.

And then by not checking if edits are empty, we can also send over an empty object in the case of only having defaults.

Reply:
Makes perfect sense 👍 however it seems like sending an empty object for relationships with junction table is only creating the junction entry, not the related collection entry:

https://user-images.githubusercontent.com/42867097/226595694-3616b9e8-4596-4cfb-a504-e2885b575bae.mp4

<br />

I tried editing the raw value and change `{}` to `{"b_id": {}}` which seems to work as expected to create a new B item with default values. I assume that means we need to make it add the related collection junction field like I tested, if I'm not mistaken.